tests/test_number.py

```diff
@@ -0,0 +1,409 @@
+import pytest
+
+from eth1spec.number import Uint
+
+
+def test_uint_new() -> None:
+    value = Uint(5)
+    assert isinstance(value, int)
+    assert isinstance(value, Uint)
+    assert value == 5
+
+
+def test_uint_new_negative() -> None:
+    with pytest.raises(ValueError):
+        Uint(-5)
+
+
+def test_uint_new_float() -> None:
+    with pytest.raises(TypeError):
+        Uint(0.1)  # type: ignore
+
+
+def test_uint_radd() -> None:
+    value = 4 + Uint(5)
+    assert isinstance(value, Uint)
+    assert value == 9
+
+
+def test_uint_radd_negative() -> None:
+    with pytest.raises(ValueError):
+        (-4) + Uint(5)
+
+
+def test_uint_radd_float() -> None:
+    value = (1.0) + Uint(5)
+    assert not isinstance(value, int)
```

Comment:
We're asserting that has been coerced to a float? Do we want to allow a float value or should this also throw?

Reply:
What I'm really asserting is that the result isn't an `int`, so the type checker should catch it.